operators/src/mock/mock_raster_source.rs

```diff
@@ -18,14 +21,28 @@ where
     T: Pixel,
 {
     pub data: Vec<RasterTile2D<T>>,
+    pub no_data_value: Option<T>,
+    pub tiling_specification: TilingSpecification,
 }
 
 impl<T> MockRasterSourceProcessor<T>
 where
     T: Pixel,
 {
-    fn new(data: Vec<RasterTile2D<T>>) -> Self {
-        Self { data }
+    fn new(
+        data: Vec<RasterTile2D<T>>,
+        no_data_value: Option<T>,
+        tiling_specification: TilingSpecification,
+    ) -> Self {
+        // check that all tiles have the size of the used tiling specification
+        data.iter()
+            .for_each(|f| assert_eq!(f.grid_shape(), tiling_specification.grid_shape()));
```

Comment:
I guess it is only used in tests, but in fact, you can easily use it in any workflow and then easily let our system panic. So it should be a `Result`.